dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -192,7 +192,7 @@ CRecycleBinEnum::~CRecycleBinEnum()
 HRESULT WINAPI CRecycleBinEnum::Initialize(DWORD dwFlags)
 {
     WCHAR szDrive[8];
-    if (GetEnvironmentVariableW(L"SystemDrive", szDrive, _countof(szDrive)))
+    if (GetEnvironmentVariableW(L"SystemDrive", szDrive, _countof(szDrive) - 1))
```

Comment:
More explicitly, like:
```suggestion
    if (GetEnvironmentVariableW(L"SystemDrive", szDrive, _countof(szDrive) - _countof(L"\")))
```

Moreover, should we foolproof the result check: `0`, `< size`, `> size`?

Reply:
_countof(L"\") won't compile (also if it's for what you thought, it's equal to 2.)